boehmgc: refactor

AndersonTorres opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- Add myself as maintainer
- Add some options


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

risicle commented:
Not sure the branch change went right.

AndersonTorres commented:
100% sure the branch change failed miserably. Fixing.

bobby285271 commented:
@ofborg build guile

bobby285271 reviewed (dismissed):
guile still fails (cherry-picked to master and tested native build, x86_64-linux), note that this is [experimental release](https://github.com/ivmai/bdwgc/commit/47e9106c17b72e9ee5501308f69ea94531e798b3) and previously reverted in https://github.com/NixOS/nixpkgs/pull/152145#issuecomment-1001133645


<details>

```
Making all in libguile
make[2]: Entering directory '/build/guile-2.2.7/libguile'
  GEN      libpath.h
  SNARF  alist.x
In file included from ../libguile/bdw-gc.h:47,
                 from ../libguile/gc.h:156,
                 from ../libguile/pairs.h:28,
                 from ../libguile/_scm.h:74,
                 from alist.c:25:
/nix/store/iw749gq7s1szysr688llqc65d2k2r8a5-boehm-gc-8.2.0-dev/include/gc/gc.h:1784:11: fatal error: gc_pthread_redirects.h: No such file or directory
 1784 | # include "gc_pthread_redirects.h"
      |           ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:4264: alist.x] Error 1
make[2]: Leaving directory '/build/guile-2.2.7/libguile'
make[1]: *** [Makefile:1857: all-recursive] Error 1
make[1]: Leaving directory '/build/guile-2.2.7'
make: *** [Makefile:1743: all] Error 2
```

</details>

AndersonTorres commented:
Being experimental, I will remove the "upgrade" commit.

AndersonTorres committed:
boehmgc: refactor

- use finalAttrs-style recursive attributes
- create option for mmap
- add myself as maintainer

AndersonTorres commented:
@ofborg build guile

bobby285271 commented:
May target master if there is no rebuilds. I am not familiar with this pkg, ask for someone else if you need a review :-)

AndersonTorres commented:
@ofborg build guile

AndersonTorres commented:
@ofborg build guile

Mindavi commented on pkgs/development/libraries/boehm-gc/default.nix:
```diff
@@ -22,22 +27,22 @@ stdenv.mkDerivation rec {
+  nativeBuildInputs = lib.optional stdenv.hostPlatform.isMinGW autoreconfHook;
 
-  doCheck = true; # not cross;
+  doCheck = true; # not cross
```
Nit: I think 'not cross' is implied, no need to keep it here

Mindavi commented:
Diff LGTM, have you tried building and running nix with this?

AndersonTorres commented:
@ofborg build guile nix

AndersonTorres commented:
@ofborg build guile nix

jonringer commented on pkgs/development/libraries/boehm-gc/default.nix:
```diff
@@ -54,13 +59,12 @@ stdenv.mkDerivation rec {
   };
-}
+
+  passthru.tests = nixVersions;
```
not sure how I feel about passthru being after meta. Personally, i don't expect there to be anything after meta, has this changed?

AndersonTorres replied:
To be honest I don't know. I found both ways in a cursory look.
I believe that both are a bit "meta" in the sense they do not trigger rebuilds.
I will revert this.

Mindavi reviewed (changes requested):
One comment, otherwise looks ok. Can be retargeted to master again as long as nothing is changed that causes new rebuilds :)

AndersonTorres commented on pkgs/development/libraries/boehm-gc/default.nix:
```diff
@@ -1,19 +1,24 @@
 , autoreconfHook
-, enableLargeConfig ? false # doc: https://github.com/ivmai/bdwgc/blob/v8.0.6/doc/README.macros (LARGE_CONFIG)
+# doc: https://github.com/ivmai/bdwgc/blob/v8.0.6/doc/README.macros (LARGE_CONFIG)
+, enableLargeConfig ? config.boehmgc.enableLargeConfig or false
```
The idea is to "override" the default value. Something wrong with this?

bobby285271 commented:
@ofborg eval

SuperSandro2000 commented:
@ofborg eval

AndersonTorres commented:
Any noticeable reason for the egregious failure on Darwintel?

AndersonTorres commented:
@ofborg build guile nix
